src/ui/public/visualize/loader/embedded_visualize_handler.js

```diff
@@ -26,58 +30,144 @@ const RENDER_COMPLETE_EVENT = 'render_complete';
  * with the visualization.
  */
 export class EmbeddedVisualizeHandler {
-  constructor(element, scope, savedObject) {
+  constructor(element, savedObject, params) {
     this._element = element;
-    this._scope = scope;
     this._savedObject = savedObject;
+    this._params = {
+      uiState: params.uiState,
+      queryFilter: params.queryFilter,
+      searchSource: savedObject.searchSource,
+      aggs: savedObject.vis.getAggConfig(),
+      timeRange: params.timeRange,
+      filters: params.filters,
+      query: params.query,
+    };
```

Comment:
Small suggestion: how about destructuring here, to make it clearer what we care about in the `savedObject` and `params`?

```js
    const { searchSource, vis } = savedObject;

    const {
      uiState,
      queryFilter,
      timeRange,
      filters,
      query,
    } = params;

    this._params = {
      uiState: uiState,
      queryFilter: queryFilter,
      searchSource: searchSource,
      aggs: vis.getAggConfig(),
      timeRange: timeRange,
      filters: filters,
      query: query,
    };
```